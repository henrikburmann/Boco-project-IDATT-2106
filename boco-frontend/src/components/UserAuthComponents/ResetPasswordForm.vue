<template>
  <div
    class="md:ring-1 ring-gray-300 rounded-xl overflow-hidden mx-auto mb-auto max-w-md w-full p-4"
  >
    <h3 class="text-xl font-medium text-center text-primary-light mt-4 mb-8">
      Glemt passordet ditt?
    </h3>

    <div
      id="emailField"
      class="m-6"
      :class="{ error: v$.email.$errors.length }"
    >
      <div class="mb-6">
        <label
          for="email"
          class="block mb-2 text-sm font-medium text-gray-900 dark:text-gray-300"
          >E-post</label
        >
        <input
          type="email"
          id="email"
          class="block w-full px-4 py-2 mt-2 text-gray-700 placeholder-gray-500 bg-white border rounded-md dark:bg-gray-800 dark:border-gray-600 dark:placeholder-gray-400 focus:border-blue-400 dark:focus:border-blue-300 focus:ring-opacity-40 focus:outline-none focus:ring focus:ring-blue-300"
          placeholder="eksempel@eksempel.no"
          v-model="v$.email.$model"
          required
          @keyup.enter="sendHome"
        />
        <!-- error message -->
        <div v-for="(error, index) of v$.email.$errors" :key="index">
          <div
            class="text-error-medium text-sm"
            v-show="showError"
            id="emailErrorId"
          >
            {{ error.$message }}
          </div>
        </div>
      </div>
      <Button
        class="float-right"
        @click="sendHome"
        :text="'Tilbakestill passord'"
      />
    </div>
  </div>
</template>

<script>
import useVuelidate from "@vuelidate/core";
import { required, email, helpers } from "@vuelidate/validators";
import Button from "@/components/BaseComponents/ColoredButton";

export default {
  name: "ResetPassword.vue",

  components: {
    Button,
  },
  data() {
    return {
      showError: false,
      email: "",
    };
  },
  setup() {
    return { v$: useVuelidate() };
  },
  validations() {
    return {
      email: {
        required: helpers.withMessage("Feltet må være utfylt", required),
        email: helpers.withMessage(`E-posten er ugyldig`, email),
      },
    };
  },
  methods: {
    sendHome() {
      this.showError = true;

      this.v$.email.$touch();

      if (this.v$.email.$invalid) {
        return;
      } else {
        this.$router.push("/");
      }
    },
  },
  validate() {
    this.$refs.form.validate();
  },
};
</script>
