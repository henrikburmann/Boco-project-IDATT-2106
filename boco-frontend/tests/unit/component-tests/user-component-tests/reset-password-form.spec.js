import { mount } from "@vue/test-utils";
import ResetPasswordForm from "@/components/UserAuthComponents/ResetPasswordForm.vue";

describe("ResetPasswordForm component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(ResetPasswordForm);
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("is instantiated", () => {
    expect(wrapper.exists()).toBeTruthy();
  });
});
