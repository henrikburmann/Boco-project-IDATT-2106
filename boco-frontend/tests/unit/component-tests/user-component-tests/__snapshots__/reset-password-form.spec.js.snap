// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPasswordForm component renders correctly 1`] = `
<div
  class="md:ring-1 ring-gray-300 rounded-xl overflow-hidden mx-auto mb-auto max-w-md w-full p-4"
>
  <h3
    class="text-xl font-medium text-center text-primary-light mt-4 mb-8"
  >
     Glemt passordet ditt? 
  </h3>
  <div
    class="m-6"
    id="emailField"
  >
    <div
      class="mb-6"
    >
      <label
        class="block mb-2 text-sm font-medium text-gray-900 dark:text-gray-300"
        for="email"
      >
        E-post
      </label>
      <input
        class="block w-full px-4 py-2 mt-2 text-gray-700 placeholder-gray-500 bg-white border rounded-md dark:bg-gray-800 dark:border-gray-600 dark:placeholder-gray-400 focus:border-blue-400 dark:focus:border-blue-300 focus:ring-opacity-40 focus:outline-none focus:ring focus:ring-blue-300"
        id="email"
        placeholder="eksempel@eksempel.no"
        required=""
        type="email"
      />
      <!-- error message -->
      
      
    </div>
    
    <!-- Button with custom text and color -->
    <button
      class="flex items-center px-2 py-2 font-medium tracking-wide capitalize text-white transition-colors duration-200 transform rounded-md focus:outline-none focus:ring focus:ring-opacity-80 min-w-{20px} bg-primary-medium hover:bg-primary-dark focus:ring-primary-light float-right"
    >
      Tilbakestill passord
    </button>
    
  </div>
</div>
`;
